dev-packages/application-package/src/generator/frontend-generator.ts

```diff
@@ -87,25 +87,46 @@ if (cluster.isMaster) {
     }
     process.env.LC_NUMERIC = 'C';
     const electron = require('electron');
-    const path = require('path');
     electron.app.on('window-all-closed', function () {
         if (process.platform !== 'darwin') {
             electron.app.quit();
         }
     });
     electron.app.on('ready', function () {
+        const path = require('path');
+        const { fork }  = require('child_process');
+        // Check whether we are in bundled application or development mode.
+        const devMode = process.defaultApp || /node_modules[\\/]electron[\\/]/.test(process.execPath);
         const mainWindow = new electron.BrowserWindow({ width: 1024, height: 728 });
-        require("../backend/main").then(address => {
-            mainWindow.loadURL(\`file://\${path.join(__dirname, '../../lib/index.html')}?port=\${address.port}\`);
-        }).catch(() => {
-            electron.app.exit(1);
-        });
+        const mainPath = path.join(__dirname, '..', 'backend', 'main');
+        const loadMainWindow = function(port) {
+            mainWindow.loadURL(\`file://\${path.join(__dirname, '../../lib/index.html')}?port=\${port}\`);
+        };
+        // We need to distinguish between bundled application and development mode when starting the clusters.
+        // https://github.com/electron/electron/issues/6337#issuecomment-230183287
+        if (devMode) {
+            require(mainPath).then(address => {
+                loadMainWindow(address.port);
```

Comment:
Does it still work? It looks like `main.js` does not export an instance of address anymore, see: https://github.com/theia-ide/theia/pull/746/files#r147333752

Reply:
It did work. Anyway, I do another check.

Reply:
Fixed with 1a3e09c0a77d2650a1036eafaa087543b8526da7.